core/src/main/scala/org/ensime/core/Analyzer.scala

```diff
@@ -174,90 +223,130 @@ class Analyzer(
     case TypecheckFilesReq(files) =>
       sender ! handleReloadFiles(files.map(SourceFileInfo(_)))
     case req: PrepareRefactorReq =>
-      sender ! handleRefactorPrepareRequest(req)
+
+      sender ! scalaCompiler.map(handleRefactorPrepareRequest(_, req)).getOrElse(VoidResponse)
     case req: ExecRefactorReq =>
-      sender ! handleRefactorExec(req)
+      sender ! scalaCompiler.map(handleRefactorExec(_, req)).getOrElse(VoidResponse)
     case req: CancelRefactorReq =>
       sender ! handleRefactorCancel(req)
+    case CompletionsReq(f, point, maxResults, caseSens, reload) if FileUtils.isJava(f.file) =>
+      javaCompiler.askCompletionsAtPoint(f, point, maxResults, caseSens) pipeTo sender
```

Comment:
these uses of `pipeTo` actually end up returning the wrong type. They forward the `Option` whereas we have been returning the value or `FalseResponse` until now. I wonder if perhaps we should just leave `Analyzer` alone and then separate out the Java requests in `Project` and send to a Java actor. We'd have to be careful when a request has multiple files in the parameter, which are mixed language, but that is probably never valid anyway.